core/block_validator.go

```diff
@@ -137,6 +140,52 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	return nil
 }
 
+// ValidateSystemTxs validates all system txs contained in a block.
+// We check that:
+// - the sender is a whitelisted address
+// - the recipient is a system contract
+func (v *BlockValidator) ValidateSystemTxs(block *types.Block) error {
```

Comment:
So any transaction-payload-specific validation will happen in the target contract, right?

Reply:
yeah, nothing to do here -- we are agnostic to the contract details.